controllers/aws/s3-bucket.go

```diff
@@ -78,13 +81,18 @@ func (a *AwsClient) HandleBucketDeletion(bucketToDelete string) (bool, error) {
 
 func (a *AwsClient) HandleBucketUpdate(bucketName string, bucketSpec *s3operatorv1.S3BucketSpec) error {
 	a.Log.V(1).Info("HandleBucketUpdate function")
-	_, err := a.updateBucketTags(bucketName, bucketSpec.Tags)
-
+	isOwner, err := a.isBucketManagedByOperator(bucketName)
+	if isOwner {
+		_, err = a.updateBucketTags(bucketName, bucketSpec.Tags)
+	} else {
```

Comment:
why not like this:
```
else if err == nil {
   err = errors.New("cant update bucket that not manage by operator");
}
```
always better to avoid nested if as much as possible (for maintainability reson)